### ISSUE TITLE
Improvement: automatically generate UUID for azurerm_role_assignment

### ISSUE THREAD

**Issue by jansepke** [terraform-providers/terraform-provider-azurerm#667]
### Terraform Version
Terraform v0.11.1
+ provider.azurerm v1.0.0

### Affected Resource(s)
azurerm_role_assignment

### Terraform Configuration Files
```hcl
resource "azurerm_role_assignment" "test" {
  name               = "1a679e7d-719c-42bf-9141-15c8de83836d" // random UUID
  scope              = "..."
  role_definition_id = "..."
  principal_id       = "..."
}
```

Hi there,

I don't understand why I need to specify a unique UUID for a azurerm_role_assignment resource. This makes this piece of code less modular. Would it be possible to make this property optional and fall back to an automatically generated UUID? Or is there any way to generate a static UUID in terraform? When I use the `${uuid()}` interpolation, terraform will recreate the azurerm_role_assignment each run because the interpolation has changed.

**Comment by tombuildsstuff** [comment]
hey @jansepke 

Thanks for opening this issue, I've added support for this in #685 which has been merged and will be released as part of v1.0.2 of the AzureRM Provider.

Thanks!

**Comment by JamesDLD** [comment]
Thanks for this improvement, just raised the same improvement here #1371 for the resource **role_definition_id**
